mssql-plugin/src/e2e-test/java/io.cdap.plugin/mssql/BQValidation.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.mssql;
+
+import com.google.cloud.bigquery.TableResult;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import io.cdap.e2e.utils.BigQueryClient;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+import io.cdap.plugin.MssqlClient;
+import org.apache.spark.sql.types.Decimal;
+import org.junit.Assert;
+import stepsdesign.BeforeActions;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+import java.util.TimeZone;
+
+/**
+ * BQValidation.
+ */
+public class BQValidation {
+  /**
+   * Extracts entire data from source and target tables.
+   *
+   * @param sourceTable table at the source side
+   * @param targetTable table at the sink side
+   * @return true if the values in source and target side are equal
+   */
+
+  public static boolean validateDBToBQRecordValues(String schema, String sourceTable, String targetTable)
+    throws SQLException, ClassNotFoundException, IOException, InterruptedException, ParseException {
+    List<Object> targetBigQueryRows = new ArrayList<>();
+    List<JsonObject> targetBigQueryResponse = new ArrayList<>();
+
+    getBigQueryTableData(targetTable, targetBigQueryRows);
+
+    for (Object rows : targetBigQueryRows) {
+      JsonObject json = new Gson().fromJson(String.valueOf(rows), JsonObject.class);
+      targetBigQueryResponse.add(json);
+    }
+
+    String getSourceQuery = "SELECT * FROM " + schema + "." + sourceTable;
+    try (Connection connect = MssqlClient.getMssqlConnection()) {
+      connect.setHoldability(ResultSet.HOLD_CURSORS_OVER_COMMIT);
+      Statement statement1 = connect.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE,
+                                                     ResultSet.HOLD_CURSORS_OVER_COMMIT);
+
+      ResultSet rsSource = statement1.executeQuery(getSourceQuery);
```

Comment:
nit: This 6 lines of code is repetitive. we can move this to a function which takes query and response as inputs and returns a result set.

Reply:
Done.